ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -292,19 +276,54 @@ private void handleNewBlockHashesFromNetwork(final EthMessage message) {
       return CompletableFuture.completedFuture(block);
     }
 
-    ethContext.getScheduler().scheduleSyncWorkerTask(() -> validateAndBroadcastBlock(block));
+    final BlockHeader parent =
+        protocolContext
+            .getBlockchain()
+            .getBlockHeader(block.getHeader().getParentHash())
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Incapable of retrieving header from non-existent parent of "
+                            + block.getHeader().getNumber()
+                            + "."));
 
-    // Import block
-    final PersistBlockTask<C> importTask =
-        PersistBlockTask.create(
-            protocolSchedule, protocolContext, block, HeaderValidationMode.FULL, metricsSystem);
+    final ProtocolSpec<C> protocolSpec =
+        protocolSchedule.getByBlockNumber(block.getHeader().getNumber());
+    final BlockHeaderValidator<C> blockHeaderValidator = protocolSpec.getBlockHeaderValidator();
     return ethContext
         .getScheduler()
-        .scheduleSyncWorkerTask(importTask::run)
+        .scheduleSyncWorkerTask(
+            () -> validateAndProcessPendingBlock(blockHeaderValidator, block, parent));
+  }
+
+  private CompletableFuture<Block> validateAndProcessPendingBlock(
+      final BlockHeaderValidator<C> blockHeaderValidator,
+      final Block block,
+      final BlockHeader parent) {
+    if (blockHeaderValidator.validateHeader(
+        block.getHeader(), parent, protocolContext, HeaderValidationMode.FULL)) {
+      ethContext.getScheduler().scheduleSyncWorkerTask(() -> broadcastBlock(block, parent));
+      return ethContext.getScheduler().scheduleSyncWorkerTask(() -> runImportTask(block));
```

Comment:
You don't need to defer this to a separate worker task since you're already in a worker thread.
```suggestion
      return runImportTask(block);
```